contracts/wallet.sol

```diff
@@ -704,79 +707,56 @@ contract Wallet is ENSResolvable, Vault, GasTopUpLimit, LoadLimit {
         return _amount.mul(stablecoinMagnitude).div(stablecoinRate);
     }
 
-    /// @dev This function is taken from the Gnosis MultisigWallet: https://github.com/gnosis/MultiSigWallet/
-    /// @dev License: https://github.com/gnosis/MultiSigWallet/blob/master/LICENSE
-    /// @dev thanks :)
     /// @dev This calls proxies arbitrary transactions to addresses
-    /// @param _destination address of the transaction
+    /// @param _to desgtination address of the transaction
     /// @param _value ETH amount in wei
-    /// @param _dataLength length of the transaction data
     /// @param _data transaction payload binary
-    // call has been separated into its own function in order to take advantage
-    // of the Solidity's code generator to produce a loop that copies tx.data into memory.
-    function _externalCall(address _destination, uint _value, uint _dataLength, bytes _data) private returns (bool) {
-        bool result;
-        assembly {
-            let x := mload(0x40)   // "Allocate" memory for output (0x40 is where "free memory" pointer is stored by convention)
-            let d := add(_data, 32) // First 32 bytes are the padded length of data, so exclude that
-            result := call(
-                sub(gas, 34710),   // 34710 is the value that solidity is currently emitting
-                                   // It includes callGas (700) + callVeryLow (3, to pay for SUB) + callValueTransferGas (9000) +
-                                   // callNewAccountGas (25000, in case the destination address does not exist and needs creating)
-                _destination,
-                _value,
-                d,
-                _dataLength,        // Size of the input (in bytes) - this is what fixes the padding problem
-                x,
-                0                  // Output is ignored, therefore the output size is zero
-            )
-        }
-
-        return result;
+    function _executeCall(address _to, uint256 _value, bytes memory _data) internal  returns (bool success)
+{
+    // solium-disable-next-line security/no-inline-assembly
+    assembly {
+        success := call(gas, _to, _value, add(_data, 0x20), mload(_data), 0, 0)
     }
+}
+
 
     /// @dev This function converts to an address
     /// @param _bts bytes
     /// @param _from start position
-    function _bytesToAddress(bytes _bts, uint _from) private pure returns (address) {
+    function _bytesToAddress(bytes memory _bts, uint _from) private pure returns (address) {
```

Comment:
Where do we use this function (why did we add it)?
Could these type conversion functions be in a library instead of inside the wallet contract?

Reply:
this is actually a good point (move to a library), that we discussed before. most of them are taken from stackOverflow ( i was pretty sure we had a disclaimer)